crates/primitives/src/compression/mod.rs

```diff
@@ -38,6 +40,38 @@ thread_local! {
         ));
 }
 
+/// Fn creates as many tx compressors as times its called. It is meant to be used in no-std config
+pub fn create_tx_compressor() -> RefCell<Compressor<'static>> {
```

Comment:
no, they're not meant to be used that way. each thread call any of these fns to create its own instance of compressor, decompressor

Reply:
It's recommended to have compressor, decompressor per thread. Static is a single instance in global scope for program's runtime. I thought about implementing map that keeps instances of compressor/decompressor per thread but didn't find how the same instance of compressor and decompressor can be reused within a thread.

Reply:
Agree about RefCell, I'll make changes and retest these fns. Btw, in case of std config I can't see data race for compressor. Compressor is instantiated and used within same thread so RefCell usage is confusing.

Reply:
well, creator fn returns compressor object which is mutable due to mut binding. did you mean any changes?